azure/functions/_http_wsgi.py

```diff
@@ -0,0 +1,174 @@
+from typing import Callable, Dict, List, Optional, Any
+from io import BytesIO, StringIO
+from os import linesep
+from urllib.parse import urlparse
+from wsgiref.headers import Headers
+
+from ._abc import Context
+from ._http import HttpRequest, HttpResponse
+from ._thirdparty.werkzeug._compat import string_types, wsgi_encoding_dance
+
+
+class WsgiRequest:
+    _environ_cache: Dict[str, Any] = None
+
+    def __init__(self,
+                 func_req: HttpRequest,
+                 func_ctx: Optional[Context] = None):
+        url = urlparse(func_req.url)
+        func_req_body = func_req.get_body() or b''
+
+        # Convert function request headers to lowercase header
+        self._lowercased_headers = {
+            k.lower(): v for k, v in func_req.headers.items()
+        }
+
+        # Implement interfaces for PEP 3333 environ
+        self.request_method = getattr(func_req, 'method', None)
+        self.script_name = ''
+        self.path_info = getattr(url, 'path', None)
+        self.query_string = getattr(url, 'query', None)
+        self.content_type = self._lowercased_headers.get('content-type')
+        self.content_length = str(len(func_req_body))
+        self.server_name = getattr(url, 'hostname', None)
+        self.server_port = str(self._get_port(url, self._lowercased_headers))
```

Comment:
`getattr` might be slightly slower (due to reflection) than regular get, but may not be a big deal. Just a note.